### ISSUE TITLE
[Feature request] Provide API to set custom ClassInstanceManager

### ISSUE THREAD

**Issue by mdolinin** [getgauge/gauge-java#89]
[Feature request] Provide API to set custom ClassInstanceManager.
This is needed for proper integration with IoC frameworks (Spring, Guice, Dagger, PicoContainer, etc...)
Also described here #49

**Comment by kashishm** [comment]
@mdolinin You can use `ClassInstanceManager.setClassInitializer()` for this. Here's an [example usage](https://github.com/getgauge/gauge-example-spring/blob/master/src/test/java/RegisterIOC.java#L12).

**Comment by mdolinin** [comment]
@kashishm But i couldn't control Hooks instances.
Second when i run in parallel mode. This solution will create new Spring context for each thread.
But what if we want to use one context for all tests, and share some objects.

**Comment by kashishm** [comment]
> But i couldn't control Hooks instances.

I am not sure if I understand this. Can you please elaborate?
> Second when i run in parallel mode. This solution will create new Spring context for each thread.
But what if we want to use one context for all tests, and share some objects.

 In multi process parallel execution, creating new Spring context for each process is required and in multi thread parallel execution you can use singleton `ClassInitializer` instance and use it in `ClassInstanceManager.setClassInitializer()` for every thread.

**Comment by RabihHallage** [comment]
Since issue #49 is closed, this issue should be closed as well

**Comment by kashishm** [comment]
Duplicate of #49.
